scipy/stats/_hypotests.py

```diff
@@ -663,3 +665,241 @@ def somersd(x, y=None):
         raise ValueError("x must be either a 1D or 2D array")
     d, p = _somers_d(table)
     return SomersDResult(d, p, table)
+
+
+def _barnard_exact_compute_gammaln_combination(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = gammaln(np.arange(n + 1) + 1)
+    return gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+BarnardExactResult = make_dataclass("BarnardExactResult", [
+    ('statistic', float),
+    ("pvalue", float)
+])
+
+
+def barnard_exact(
+        table,
+        alternative='two-sided',
+        pooled=True,
+        num_it=3):
+    """
+    Perform a Barnard exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': one-sided
+          * 'greater': one-sided
```

Comment:
```suggestion
        * 'two-sided'
        * 'less': one-sided
        * 'greater': one-sided
```
No additional indent needed, otherwise it renders as a quotation.